### PR TITLE
countdown content type

### PR DESCRIPTION
**Pull request by cspruiell**

### CODE REVIEW

File: Example/PageBuilderCountdown/view/adminhtml/layout/pagebuilder_example_countdown_form.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0"?>
+<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" layout="admin-1column" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
```
Comment (davemacaulay) [inline Example/PageBuilderCountdown/view/adminhtml/layout/pagebuilder_example_countdown_form.xml:8]: You're missing copyright in a few of these files

File: Example/PageBuilderCountdown/view/adminhtml/web/js/content-type/example-countdown/preview.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    Preview.prototype = Object.create(PreviewBase.prototype);
+
+    /**
+     * Populate the countdown timer with values based on the target date
+     *
+     * @returns {void}
+     */
+    Preview.prototype.populateCountdown = function populateCountdown(preview) {
+        let target = '';
+        let countdownField = document.querySelector(`#${preview.contentType.id} .countdown-field`);
```
Comment (davemacaulay) [inline Example/PageBuilderCountdown/view/adminhtml/web/js/content-type/example-countdown/preview.js]: We can't use backticks in JS (due to IE), but this is just an example so don't think we need to update it

File: Example/PageBuilderCountdown/view/base/web/js/content-type/example-countdown/appearance/default/widget.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+        intervalId = setInterval(function (field, intervalId) {
+            let target = '';
+            let countdownField = field[0].children[0];
+
+            if (countdownField) {
+                target = countdownField.getAttribute('target_date');
+            }
+
+            if (!target || target === '') {
```
Comment (davemacaulay) [inline Example/PageBuilderCountdown/view/base/web/js/content-type/example-countdown/appearance/default/widget.js]: You have this logic duplicated across the admin and the store front, you could write a function within base and import it into both areas to save on duplicating code.

**Review by davemacaulay — COMMENTED** [review]
Some minor things that'd be good to fix

File: Example/PageBuilderCountdown/view/adminhtml/web/js/content-type/example-countdown/preview.js (hunk 2 above)
Comment (davemacaulay) [reply Example/PageBuilderCountdown/view/adminhtml/web/js/content-type/example-countdown/preview.js]: @cspruiell  you didn't resolve this

File: Example/PageBuilderCountdown/view/adminhtml/web/js/content-type/example-countdown/preview.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+// import setCountdownField from 'Magento_PageBuilderCountdown/js/content-type/example-countdown/countdown';
```
Comment (davemacaulay) [inline Example/PageBuilderCountdown/view/adminhtml/web/js/content-type/example-countdown/preview.js]: We don't need this comment

File: Example/PageBuilderCountdown/view/adminhtml/web/js/content-type/example-countdown/preview.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+        if (!countdownField) {
+            return;
+        }
+
+        if (preview && preview.data && preview.data.countdown_target) {
+            target = preview.data.countdown_target.attributes()['target_date'];
+        }
+
+        this.setCountdownField(target, countdownField, this.intervalId);
```
Comment (davemacaulay) [inline Example/PageBuilderCountdown/view/adminhtml/web/js/content-type/example-countdown/preview.js:45]: Why are we calling this function with this? It seems like we're importing it as `setCountdownField`

File: Example/PageBuilderCountdown/view/base/web/js/content-type/example-countdown/appearance/default/widget.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+        intervalId = setInterval(function (field, intervalId) {
+            let target = '';
+            let countdownField = field[0].children[0];
+
+            if (countdownField) {
+                target = countdownField.getAttribute('target_date');
+            }
+
+            this.setCountdownField(target, countdownField, intervalId);
```
Comment (davemacaulay) [inline Example/PageBuilderCountdown/view/base/web/js/content-type/example-countdown/appearance/default/widget.js:24]: Same here, we shouldn't have this in front.

File: Example/PageBuilderCountdown/view/base/web/js/content-type/example-countdown/countdown.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+/**
+ * Sets the countdown fields with remaining time
+ */
+function setCountdownField(target, countdownField, intervalId) {
```
Comment (davemacaulay) [inline Example/PageBuilderCountdown/view/base/web/js/content-type/example-countdown/countdown.js:13]: A JSDoc comment here would help illustrate what to pass into each parameter

File: Example/PageBuilderCountdown/view/adminhtml/web/js/content-type/example-countdown/preview.js (hunk 2 above)
Comment (cspruiell) [reply Example/PageBuilderCountdown/view/adminhtml/web/js/content-type/example-countdown/preview.js]: thought you were saying not to fix it but, fixed! ;p

File: Example/PageBuilderCountdown/view/base/web/js/content-type/example-countdown/appearance/default/widget.js (hunk 6 above)
Comment (cspruiell) [reply Example/PageBuilderCountdown/view/base/web/js/content-type/example-countdown/appearance/default/widget.js:24]: this is that weird thing we talked about where it doesn't work without "this." in front of the function name. (I accidentally removed it from the backend file but have restored it.)

File: Example/PageBuilderCountdown/view/base/web/js/content-type/example-countdown/countdown.js (hunk 7 above)
Comment (cspruiell) [reply Example/PageBuilderCountdown/view/base/web/js/content-type/example-countdown/countdown.js:13]: done

**Comment by bdenham** [comment]
Will look at this and update any compatibility problems soonish. :)

**Review by bdenham — APPROVED** [review]
Adding this — Finally! If it doesn't work, I'm soliciting PR fixes from interested parties. Thanks!

**Pull request merged**